modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -964,23 +967,30 @@ public void stop() {
         metaStorageMgr.unregisterWatch(stableAssignmentsRebalanceListener);
```

Comment:
beforeNodeStop was added in order to have an ability to finish some actions that require local network layer to be up and running.
Why do we call e.g. metaStorageMgr.unregisterWatch() within beforeNodeStop? Does it really need a network? Or it should go before some other action that requires network or ...? It's also about stopping tables. Does it required a network?

Reply:
> Why do we call e.g. metaStorageMgr.unregisterWatch() within beforeNodeStop? Does it really need a network?

I've put these calls here, so that meta storage watches would be disabled when we are calling `stop`. Does it make sense?

> It's also about stopping tables. Does it required a network?

For tables - yes, we need the network. This is what the original problem of this ticket was about: when we call `TableManager#stop` we also call `Replica#shutdown` which does a network call (which never succeeds). That's why I moved the whole `cleanupTableResources` to `beforeNodeStop`.

Reply:
> I've put these calls here, so that meta storage watches would be disabled when we are calling stop. Does it make sense?

That doesn't matter, but because ms.unregister should go before replica.stop and replica stop do require network, proposed order and localization seems reasonable.

> For tables - yes, we need the network. This is what the original problem of this ticket was about: when we call TableManager#stop we also call Replica#shutdown which does a network call (which never succeeds). That's why I moved the whole cleanupTableResources to beforeNodeStop.

Sounds reasonable.